extensions/prototype-content-with-issues-table/app.R

```
@@ -31,34 +32,7 @@ get_failed_job_data <- function(item, usage) {
     {
```

> [What are we catching here?](https://github.com/posit-dev/connect-extensions/pull/33#discussion_r1989916166)

We are reacting to the error returned by `connectapi` when jobs are requested for a content item that has none, like unpublished content.

---

Can we detect that there are no jobs before we hit this endpoint? If yes, we could do that here in an `if` / `else` (or we could do it down in [line 117](https://github.com/posit-dev/connect-extensions/blob/43cfec36926851d6a883187fcea3cd14f0a4761d/extensions/prototype-content-with-issues-table/app.R#L117) inside of the `bad_content_df` generation). This way, we know that it's being skipped because there are no jobs and not for some other reason that would get eaten by the try/catch.

---

Not until [Marco's PR to add a `failed` query parameter to `v1/content` and failed job data to content items](https://github.com/posit-dev/connect/pull/30414) merges.

---

Hmm, ok maybe I'm misunderstanding something then. When I tried this just now for an item on dogfood that has no jobs, I didn't get an error I got a dataframe with 0 rows (or an empty list if I used `get_job_list()`)

```
> client <- connect()
Defining Connect with server: https://rsc.radixu.com/
> item <- content_item(client, "b33cc440-f674-45a0-b5d3-7c0bb162023f")
> get_jobs(item)
# A tibble: 0 × 23
# ℹ 23 variables: id <chr>, ppid <chr>, pid <chr>, key <chr>, remote_id <chr>,
#   content_id <chr>, content_guid <chr>, app_id <chr>, app_guid <chr>,
#   variant_id <chr>, bundle_id <chr>, start_time <dttm>, end_time <dttm>,
#   last_heartbeat_time <dttm>, queued_time <dttm>, queue_name <chr>,
#   tag <chr>, exit_code <int>, status <int>, hostname <chr>, cluster <chr>,
#   image <chr>, run_as <chr>
> get_job_list(item)
list()
```

We might need to treat that differently in the code that follows here, but from what I'm seeing that shouldn't return _an error_. Could you describe / show the exact error(s) that you're catching here? 

<details>
I _also_ tried with an item that is unpublished, and that also didn't error for me, though it did actually have a job. That unpublished-but-with-a-job might be unique to the kind of failure that particular item had:

```
> unpublished_item <- content_item(client, "f8bf721e-06bf-4a69-8adf-15d6fee4bd37")
> get_jobs(unpublished_item)
# A tibble: 1 × 23
  id       ppid    pid   key   remote_id content_id content_guid app_id app_guid
  <chr>    <chr>   <chr> <chr> <chr>     <chr>      <chr>        <chr>  <chr>   
1 47369771 3261649 3261… Z93V… NA        74085      f8bf721e-06… 74085  f8bf721…
# ℹ 14 more variables: variant_id <chr>, bundle_id <chr>, start_time <dttm>,
#   end_time <dttm>, last_heartbeat_time <dttm>, queued_time <dttm>,
#   queue_name <chr>, tag <chr>, exit_code <int>, status <int>, hostname <chr>,
#   cluster <chr>, image <chr>, run_as <chr>
```
</details>

---

Perhaps our assumption that the error was due to the content item having no jobs is incorrect, and this is instead malformed content on Dogfood that is missing data we return as part of the `Content` object?

---

Did you get to the bottom of this? Or are you still investigating?

---

Is the list of malformed items on dogfood smallish? If yes, IMHO making a blocklist of content guids that we don't even try from dogfood and then not swallowing errors would be the way to go here rather than wrapping fully in a try/catch like this. It's fine if this is the way forward for now, but it is very much a code funk that we don't want to pull over into our full version of this as we transition into that part of our process.

---

I removed the `tryCatch` as 404ing content URLs were not encountered while requesting failed jobs against connect.posit.it.